avm/res/batch/batch-account/main.bicep

```diff
@@ -204,18 +204,30 @@ resource batchAccount_diagnosticSettings 'Microsoft.Insights/diagnosticSettings@
 module batchAccount_privateEndpoints 'br/public:avm/res/network/private-endpoint:0.3.1' = [for (privateEndpoint, index) in (privateEndpoints ?? []): {
   name: '${uniqueString(deployment().name, location)}-BatchAccount-PrivateEndpoint-${index}'
   params: {
-    privateLinkServiceConnections: [
+    name: privateEndpoint.?name ?? 'pep-${last(split(batchAccount.id, '/'))}-${privateEndpoint.service}-${index}'
+    privateLinkServiceConnections: privateEndpoint.?manualPrivateLinkServiceConnections != true ? [
       {
-        name: name
+        name: privateEndpoint.?privateLinkServiceConnectionName ?? '${last(split(batchAccount.id, '/'))}-${privateEndpoint.service}-${index}'
         properties: {
           privateLinkServiceId: batchAccount.id
           groupIds: [
-            privateEndpoint.?service ?? 'batchAccount'
```

Comment:
Why did we remove the default for this and other resources?
Modules with only one sub-resource type should implement schema 1 (with a default)

Reply:
These are all services with more than one sub-resource (see table) which means they need to align to schema 2.

Reply:
ah wow, ok, thanks :)